src/main/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/database/jpa/mapper/induction/InductionEntityMapper.kt

```diff
@@ -19,16 +30,79 @@ import uk.gov.justice.digital.hmpps.educationandworkplanapi.domain.induction.dto
     WorkOnReleaseEntityMapper::class,
   ],
 )
-interface InductionEntityMapper {
+abstract class InductionEntityMapper {
+
+  @Autowired
+  private lateinit var futureWorkInterestsEntityMapper: FutureWorkInterestsEntityMapper
+
+  @Autowired
+  private lateinit var inPrisonInterestsEntityMapper: InPrisonInterestsEntityMapper
+
+  @Autowired
+  private lateinit var personalSkillsAndInterestsEntityMapper: PersonalSkillsAndInterestsEntityMapper
+
+  @Autowired
+  private lateinit var previousQualificationsEntityMapper: PreviousQualificationsEntityMapper
+
+  @Autowired
+  private lateinit var previousTrainingEntityMapper: PreviousTrainingEntityMapper
+
+  @Autowired
+  private lateinit var previousWorkExperiencesEntityMapper: PreviousWorkExperiencesEntityMapper
+
+  @Autowired
+  private lateinit var workOnReleaseEntityMapper: WorkOnReleaseEntityMapper
+
   @ExcludeJpaManagedFieldsIncludingDisplayNameFields
   @GenerateNewReference
   @Mapping(target = "createdAtPrison", source = "prisonId")
   @Mapping(target = "updatedAtPrison", source = "prisonId")
-  fun fromCreateDtoToEntity(dto: CreateInductionDto): InductionEntity
+  abstract fun fromCreateDtoToEntity(dto: CreateInductionDto): InductionEntity
 
   @Mapping(target = "lastUpdatedBy", source = "updatedBy")
   @Mapping(target = "lastUpdatedByDisplayName", source = "updatedByDisplayName")
   @Mapping(target = "lastUpdatedAt", source = "updatedAt")
   @Mapping(target = "lastUpdatedAtPrison", source = "updatedAtPrison")
-  fun fromEntityToDomain(persistedEntity: InductionEntity): Induction
+  abstract fun fromEntityToDomain(persistedEntity: InductionEntity): Induction
+
+  @ExcludeJpaManagedFieldsIncludingDisplayNameFields
+  @ExcludeReferenceField
+  @Mapping(target = "createdAtPrison", ignore = true)
+  @Mapping(target = "updatedAtPrison", source = "prisonId")
+  @Mapping(target = "futureWorkInterests", expression = "java( updateFutureWorkInterests(entity, dto) )")
+  @Mapping(target = "inPrisonInterests", expression = "java( updateInPrisonInterests(entity, dto) )")
+  @Mapping(target = "personalSkillsAndInterests", expression = "java( updatePersonalSkillsAndInterests(entity, dto) )")
+  @Mapping(target = "previousQualifications", expression = "java( updatePreviousQualifications(entity, dto) )")
+  @Mapping(target = "previousTraining", expression = "java( updatePreviousTraining(entity, dto) )")
+  @Mapping(target = "previousWorkExperiences", expression = "java( updatePreviousWorkExperiences(entity, dto) )")
+  @Mapping(target = "workOnRelease", expression = "java( updateWorkOnRelease(entity, dto) )")
+  abstract fun updateEntityFromDto(@MappingTarget entity: InductionEntity, dto: UpdateInductionDto)
+
+  fun updateFutureWorkInterests(entity: InductionEntity, dto: UpdateInductionDto): FutureWorkInterestsEntity? =
+    futureWorkInterestsEntityMapper.updateEntityFromDto(entity.futureWorkInterests, dto.futureWorkInterests)
+      .let { entity.futureWorkInterests }
```

Comment:
Due to the fact that the `updateEntityFromDto()` methods in the associated classes are void, I can't call them directly from the java expression statements within the `@Mapping` annotations above. For example, this doesn't compile:

```
  @Mapping(target = "workOnRelease", expression = "java( workOnReleaseEntityMapper.updateEntityFromDto(entity.workOnRelease, dto.workOnRelease) )")
```

I'll read up more on MapStruct as it's possible I've missed something...

Reply:
It kinda makes sense though ... If you are trying to use a mapping annotation with a target attribute as you are, then the source or expression attribute needs to return a value doesnt it? Otherwise it cant map into the `target` field?